diff -ur kafka_2.13-4.0.0/bin/connect-distributed.sh kafka_2.13-4.0.0-new/bin/connect-distributed.sh
--- kafka_2.13-4.0.0/bin/connect-distributed.sh	2025-03-14 16:19:26.000000000 +0800
+++ kafka_2.13-4.0.0-new/bin/connect-distributed.sh	2025-06-23 00:05:43.174534063 +0800
@@ -23,7 +23,7 @@
 base_dir=$(dirname $0)
 
 if [ -z "$KAFKA_LOG4J_OPTS" ]; then
-    export KAFKA_LOG4J_OPTS="-Dlog4j2.configurationFile=$base_dir/../config/connect-log4j2.yaml"
+    export KAFKA_LOG4J_OPTS="-Dlog4j2.configurationFile=/etc/kafka/connect-log4j2.yaml"
 fi
 
 if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
diff -ur kafka_2.13-4.0.0/bin/connect-mirror-maker.sh kafka_2.13-4.0.0-new/bin/connect-mirror-maker.sh
--- kafka_2.13-4.0.0/bin/connect-mirror-maker.sh	2025-03-14 16:19:26.000000000 +0800
+++ kafka_2.13-4.0.0-new/bin/connect-mirror-maker.sh	2025-06-23 00:05:43.178534063 +0800
@@ -23,7 +23,7 @@
 base_dir=$(dirname $0)
 
 if [ -z "$KAFKA_LOG4J_OPTS" ]; then
-    export KAFKA_LOG4J_OPTS="-Dlog4j2.configurationFile=$base_dir/../config/connect-log4j2.yaml"
+    export KAFKA_LOG4J_OPTS="-Dlog4j2.configurationFile=/etc/kafka/connect-log4j2.yaml"
 fi
 
 if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
diff -ur kafka_2.13-4.0.0/bin/connect-standalone.sh kafka_2.13-4.0.0-new/bin/connect-standalone.sh
--- kafka_2.13-4.0.0/bin/connect-standalone.sh	2025-03-14 16:19:26.000000000 +0800
+++ kafka_2.13-4.0.0-new/bin/connect-standalone.sh	2025-06-23 00:05:43.182534062 +0800
@@ -23,7 +23,7 @@
 base_dir=$(dirname $0)
 
 if [ -z "$KAFKA_LOG4J_OPTS" ]; then
-    export KAFKA_LOG4J_OPTS="-Dlog4j2.configurationFile=$base_dir/../config/connect-log4j2.yaml"
+    export KAFKA_LOG4J_OPTS="-Dlog4j2.configurationFile=/etc/kafka/connect-log4j2.yaml"
 fi
 
 if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
diff -ur kafka_2.13-4.0.0/bin/kafka-run-class.sh kafka_2.13-4.0.0-new/bin/kafka-run-class.sh
--- kafka_2.13-4.0.0/bin/kafka-run-class.sh	2025-03-14 16:19:26.000000000 +0800
+++ kafka_2.13-4.0.0-new/bin/kafka-run-class.sh	2025-06-23 00:05:43.194534061 +0800
@@ -193,6 +193,12 @@
 done
 shopt -u nullglob
 
+# Patch for linux ENV
+if [ -d "/usr/lib64/kafka/" ] && [ -z "$CLASSPATH" ]; then
+  CLASSPATH=".:/usr/lib64/kafka/*"
+fi
+
+
 if [ -z "$CLASSPATH" ] ; then
   echo "Classpath is empty. Please build the project first e.g. by running './gradlew jar -PscalaVersion=$SCALA_VERSION'"
   exit 1
@@ -220,7 +226,7 @@
 # Log4j settings
 if [ -z "$KAFKA_LOG4J_OPTS" ]; then
   # Log to console. This is a tool.
-  LOG4J_DIR="$base_dir/config/tools-log4j2.yaml"
+  LOG4J_DIR="/etc/kafka/tools-log4j2.yaml"
   # If Cygwin is detected, LOG4J_DIR is converted to Windows format.
   (( WINDOWS_OS_FORMAT )) && LOG4J_DIR=$(cygpath --path --mixed "${LOG4J_DIR}")
   KAFKA_LOG4J_OPTS="-Dlog4j2.configurationFile=${LOG4J_DIR}"
diff -ur kafka_2.13-4.0.0/bin/kafka-server-start.sh kafka_2.13-4.0.0-new/bin/kafka-server-start.sh
--- kafka_2.13-4.0.0/bin/kafka-server-start.sh	2025-03-14 16:19:26.000000000 +0800
+++ kafka_2.13-4.0.0-new/bin/kafka-server-start.sh	2025-06-23 00:05:43.194534061 +0800
@@ -22,7 +22,7 @@
 base_dir=$(dirname $0)
 
 if [ -z "$KAFKA_LOG4J_OPTS" ]; then
-    export KAFKA_LOG4J_OPTS="-Dlog4j2.configurationFile=$base_dir/../config/log4j2.yaml"
+    export KAFKA_LOG4J_OPTS="-Dlog4j2.configurationFile=/etc/kafka/log4j2.yaml"
 fi
 
 if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
